lib/doorkeeper/oauth/token.rb

```diff
@@ -55,7 +55,9 @@ def self.from_request(request, *methods)
 
       def self.authenticate(request, *methods)
         if token = from_request(request, *methods)
-          AccessToken.by_token(token)
+          access_token = AccessToken.by_token(token)
+          access_token.revoke_previous_refresh_token! if access_token
```

Comment:
Do we know here that this token has been requested via the previous refresh token?